Installation issue: toml11 3.8 requires a C compiler

### Steps to reproduce the issue

```console
$ spack install toml11@3.8
```


### Error message

<details><summary>Error message</summary>

<pre>
==> Using cached archive: /home/spack-modules/var/spack/cache/_source-cache/archive/6a/6a3d20080ecca5ea42102c078d3415bef80920f6c4ea2258e87572876af77849.tar.gz
==> No patches needed for toml11
-- The C compiler identification is unknown
-- The CXX compiler identification is GNU 12.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - failed
-- Check for working C compiler: /home/spack-modules/opt/spack/linux-zen2/compiler-wrapper-1.0-r32kfxp26i7cdwsedcobg442ng2jnia5/libexec/spack/cc
-- Check for working C compiler: /home/spack-modules/opt/spack/linux-zen2/compiler-wrapper-1.0-r32kfxp26i7cdwsedcobg442ng2jnia5/libexec/spack/cc - broken
CMake Error at /home/spack-modules/opt/spack/linux-zen2/cmake-3.26.6-czuw5nxjy7pxgrctiuzzbluxhl74w34v/share/cmake-3.26/Modules/CMakeTestCCompiler.cmake:67 (message):
  The C compiler

    "/home/spack-modules/opt/spack/linux-zen2/compiler-wrapper-1.0-r32kfxp26i7cdwsedcobg442ng2jnia5/libexec/spack/cc"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: /tmp/testsimeon/spack-stage/spack-stage-toml11-3.8.1-moxg5ldjbhs7hllxysflanfchhpudawt/spack-build-moxg5ld/CMakeFiles/CMakeScratch/TryCompile-MlbVE3

    Run Build Command(s):/home/spack-modules/opt/spack/linux-zen2/cmake-3.26.6-czuw5nxjy7pxgrctiuzzbluxhl74w34v/bin/cmake -E env VERBOSE=1 /home/spack-modules/opt/spack/linux-zen2/gmake-4.4.1-75pazryyxx62gwcqylj452g5hcoyvynj/bin/gmake -f Makefile cmTC_29d
2b/fast && /home/spack-modules/opt/spack/linux-zen2/gmake-4.4.1-75pazryyxx62gwcqylj452g5hcoyvynj/bin/gmake  -f CMakeFiles/cmTC_29d2b.dir/build.make CMakeFiles/cmTC_29d2b.dir/build
    gmake[1]: Entering directory '/tmp/testsimeon/spack-stage/spack-stage-toml11-3.8.1-moxg5ldjbhs7hllxysflanfchhpudawt/spack-build-moxg5ld/CMakeFiles/CMakeScratch/TryCompile-MlbVE3'
    Building C object CMakeFiles/cmTC_29d2b.dir/testCCompiler.c.o
    /home/spack-modules/opt/spack/linux-zen2/compiler-wrapper-1.0-r32kfxp26i7cdwsedcobg442ng2jnia5/libexec/spack/cc    -o CMakeFiles/cmTC_29d2b.dir/testCCompiler.c.o -c /tmp/testsimeon/spack-stage/spack-stage-toml11-3.8.1-moxg5ldjbhs7hllxysflanfchhpudawt/
spack-build-moxg5ld/CMakeFiles/CMakeScratch/TryCompile-MlbVE3/testCCompiler.c
    [spack cc]: Error: SPACK_CC_* variables not set: this usually means a missing `depends_on("c", type="build")` in package.py
    gmake[1]: *** [CMakeFiles/cmTC_29d2b.dir/build.make:78: CMakeFiles/cmTC_29d2b.dir/testCCompiler.c.o] Error 1
    gmake[1]: Leaving directory '/tmp/testsimeon/spack-stage/spack-stage-toml11-3.8.1-moxg5ldjbhs7hllxysflanfchhpudawt/spack-build-moxg5ld/CMakeFiles/CMakeScratch/TryCompile-MlbVE3'
    gmake: *** [Makefile:127: cmTC_29d2b/fast] Error 2





  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:4 (project)


-- Configuring incomplete, errors occurred!
</pre></details>


### Information on your system

* **Spack:** 1.1.1
* **Builtin repo:** https://github.com/spack/spack-packages/commit/14570534a39b7261bd6f2d4ddce750773e62f96e
* **Python:** 3.10.12
* **Platform:** linux-ubuntu22.04-zen2

### Additional information

The [CMakeLists.txt](https://github.com/ToruNiina/toml11/blob/v3.8.1/CMakeLists.txt) of toml11 3.8 does not set a language. Therefore it checks for a C and C++ compiler.

Maybe older versions are also affected.

[spack-build-out.txt](https://github.com/user-attachments/files/26186530/spack-build-out.txt)

ping: @ToruNiina  @ashermancinelli

### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [x] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

--- a/repos/spack_repo/builtin/packages/toml11/package.py
+++ b/repos/spack_repo/builtin/packages/toml11/package.py
@@ -42,6 +42,7 @@ class Toml11(CMakePackage):
         "cxx_std", default="11", description="C++ standard", values=("11", "14", "17"), multi=False
     )
 
+    depends_on("c", type="build", when="@:3")
     depends_on("cxx", type="build")  # generated
 
     @when("@3.8.0:")

